src/components/flag/code/templates/FlagComponentTemplate.jsx

```diff
@@ -3,21 +3,12 @@ var classNames = require('classnames');
 
 module.exports = function(props) {
   var classes = {
-    'ui-component-flag': true,
-    'flag-left': props.left,
-    'flag-right': props.right,
-    'flag-top': props.top,
-    'flag-bottom': props.bottom,
-    'flag-primary': props.primary,
-    'flag-secondary': props.secondary,
-    'flag-success': props.success,
-    'flag-info': props.info,
-    'flag-warning': props.warning,
-    'flag-danger': props.danger,
-    'flag-sm': props.small,
-    'flag-md': props.medium,
-    'flag-lg': props.large
+    'component-flag': true,
   };
+  if(props.size) classes[props.size] = true;
+  if(props.purpose) classes[props.purpose] = true
+  if(props.position) classes[props.position] = true
```

Comment:
I can appreciate that some private companies might skew from this style. Oracle's Java style guide permits it, for instance. Given we have struggled with quality/testing in the past, I'd like to stick to more widely adopted best-practices. Coding in large groups requires stronger protections and more paranoia. FOSS groups have largely had these arguments and I'm largely inclined to follow their lead over a marketplace for rentals. :+1: